beacon_node/beacon_chain/src/graffiti_calculator.rs

```diff
@@ -0,0 +1,399 @@
+                    // More than one version implies lighthouse is connected to
+                    // an EL multiplexer. We don't support modifying the graffiti
+                    // with these configurations.
+                    warn!(
+                        self.log,
+                        "Execution Engine multiplexer detected, using default graffiti"
+                    );
+                    return default_graffiti;
+                }
+
+                let lighthouse_commit_prefix = CommitPrefix::try_from(lighthouse_version::COMMIT_PREFIX.to_string())
+                    .unwrap_or_else(|error_message| {
+                        // This really shouldn't happen but we want to definitly log if it does
+                        crit!(self.log, "Failed to parse lighthouse commit prefix"; "error" => error_message);
+                        CommitPrefix("00000000".to_string())
+                    });
+
+                engine_version.calculate_graffiti(lighthouse_commit_prefix)
+            }
+        }
+    }
+}
+
+pub fn start_engine_version_cache_refresh_service<T: BeaconChainTypes>(
+    chain: &BeaconChain<T>,
+    executor: TaskExecutor,
+) {
+    let Some(el_ref) = chain.execution_layer.as_ref() else {
+        debug!(
+            chain.log,
+            "No execution layer configured, not starting engine version cache refresh service"
+        );
+        return;
+    };
+    if matches!(
+        chain.graffiti_calculator.beacon_graffiti,
+        GraffitiOrigin::UserSpecified(_)
+    ) {
+        debug!(
+            chain.log,
+            "Graffiti is user-specified, not starting engine version cache refresh service"
+        );
+        return;
+    }
+
+    let execution_layer = el_ref.clone();
+    let log = chain.log.clone();
+    let slot_clock = chain.slot_clock.clone();
+    let epoch_duration = chain.graffiti_calculator.epoch_duration;
+    executor.spawn(
+        async move {
+            engine_version_cache_refresh_service::<T>(
+                execution_layer,
+                slot_clock,
+                epoch_duration,
+                log,
+            )
+            .await
+        },
+        "engine_version_cache_refresh_service",
+    );
+}
+
+async fn engine_version_cache_refresh_service<T: BeaconChainTypes>(
+    execution_layer: ExecutionLayer<T::EthSpec>,
+    slot_clock: T::SlotClock,
+    epoch_duration: Duration,
+    log: Logger,
+) {
+    // Preload the engine version cache after a brief delay to allow for EL initialization.
+    // This initial priming ensures cache readiness before the service's regular update cycle begins.
+    tokio::time::sleep(ENGINE_VERSION_CACHE_PRELOAD_STARTUP_DELAY).await;
+    if let Err(e) = execution_layer.get_engine_version(None).await {
+        debug!(log, "Failed to preload engine version cache"; "error" => format!("{:?}", e));
+    }
+
+    // this service should run 3/8 of the way through the epoch
+    let epoch_delay = (epoch_duration * 3) / 8;
```

Comment:
Why at 3/8?

Reply:
There are other scheduled tasks that take place at certain times in the epoch so I just picked a time that wouldn't overlap with them